### PR TITLE
Store NUX: fix redirect at setup store setup step

### PR DESCRIPTION
**Pull request by jhnstn**
As part of the Store NUX we redirect to `/store/{site}`. At that point in the flow the site is only pending transfer so just checking `isSiteAutomatedTransfer` will not work.

This adds the `hasSitePendingAutomatedTransfer` selector to make sure we keep the signup on the store flow.

**Testing**
- visit `http://calypso.localhost:3000/start/atomic-store/design-type-with-atomic-store` and create a new site with the store option.
- The signup flow should continue to the "Building your storefront" page and not redirect to `/stats/day`

### CODE REVIEW

**Comment by lamosty** [comment]
Can you check https://github.com/Automattic/wp-calypso/pull/19106? I'm already doing something similar there, using the `hasPendingAutomatedTransfer` selector.

**Comment by jhnstn** [comment]
@lamosty Either PR is fine with me. Right now the we can't test the nux so we just need to unblock the issue so we can continue testing

**Comment by jhnstn** [comment]
@lamosty  I rebased #19106 and the issue is the same.  I'm not sure what you are suggesting.
I suppose we could add `hasSitePendingAutomaticTransfer` as a prop but it seems fair to say that the prop `isAtomicSite` can capture transferred sites or sites in the transfer pipeline. 

The state of the `woocommerce/store-on-non-atomic-sites` feature flag covers the underlying issue but when we redirect to the `/store/{site}` during the NUX `isAtomicSite` _should_ be true. 

Feel free to make changes but if it test plan works then we need to get this out ASAP. we can't test the store nux on wpcalypso without this change or setting `woocommerce/store-on-non-atomic-sites` true in the wpcalypso config. 

But even with a true feature flag value I still think `isAtomicSite` is incorrect without accounting for a pending automatic transfer.

**Comment by jhnstn** [comment]
Here is an alternative PR that should unblock Store NUX testing #19176

**Comment by lamosty** [comment]
> The state of the woocommerce/store-on-non-atomic-sites feature flag covers the underlying issue but when we redirect to the /store/{site} during the NUX isAtomicSite should be true.

I don't think `isAtomicSite` should be true for any non-Atomic site, whether they are in the Store NUX or not. We've introduced the `hasSitePendingAutomatedTransfer` flag exactly for this purpose.

I'm merging #19176 and will test whether it helps.

File: client/extensions/woocommerce/app/index.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ import config from 'config';
 import DocumentHead from 'components/data/document-head';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { isSiteAutomatedTransfer } from 'state/selectors';
+import { hasSitePendingAutomatedTransfer } from 'state/selectors';
```
Comment (allendav) [inline client/extensions/woocommerce/app/index.js]: Please combine this line and the previous one in a single destructuring

File: client/extensions/woocommerce/app/index.js [hunk 2]
```diff
@@ -89,10 +95,14 @@ function mapStateToProps( state ) {
 	const canUserManageOptions =
 		( siteId && canCurrentUser( state, siteId, 'manage_options' ) ) || false;
 	const isAtomicSite = ( siteId && !! isSiteAutomatedTransfer( state, siteId ) ) || false;
+	const hasPendingAutomatedTransfer =
+		( siteId && !! hasSitePendingAutomatedTransfer( state, siteId ) ) || false;
```
Comment (lamosty) [inline client/extensions/woocommerce/app/index.js]: `hasSitePendingAutomatedTransfer` will return `null` or `false` if `siteId` is `null` or in any other bad situation so we don't need all the code around the function call.

Not a blocker. :)

**Pull request merged**